src/lib/Libutil/misc_utils.c

```diff
@@ -1520,3 +1532,221 @@ get_preemption_order(struct preempt_ordering *porder, int req, int used)
 
 	return po;
 }
+
+#ifdef WIN32
+/**
+ * @brief
+ *	Returns the current wall clock time.
+ *
+ * @return double - number of seconds.
+ */
+static double 
+get_walltime(void)
+{
+	LARGE_INTEGER time, freq;
+
+	if (QueryPerformanceFrequency(&freq) == 0)
+        	return (0);
+
+	if (QueryPerformanceCounter(&time) == 0)
+        	return (0);
+
+	return ((double)time.QuadPart / freq.QuadPart);
+}
+
+/**
+ * @brief
+ *	Returns the current processor time.
+ *
+ * @return double - number of seconds.
+ **/
+static double
+get_cputime()
+{
+	FILETIME	create_time;
+	FILETIME	exit_time;
+	FILETIME	kernel_time;
+	FILETIME	user_time;
+
+	/* The times are returned in 100-nanosecond units */
+	if (GetProcessTimes(GetCurrentProcess(), &create_time, &exit_time, &kernel_time, &user_time) == 0)
+		return (0);
+
+	if (user_time.dwLowDateTime != 0)
+		return ((double)user_time.dwLowDateTime * 0.0000001);
+	
+        return ((double)(((unsigned long long)user_time.dwHighDateTime << 32)) * 0.0000001);
+}
+
+#else
+
+/**
+ * @brief
+ *	Returns the current wall clock time.
+ *
+ * @return double - number of seconds.
+ */
+static double
+get_walltime(void)
+{
+
+	struct timeval time;
+
+	if (gettimeofday(&time, NULL) == -1)
+        	return (0);
+
+	return ((double)time.tv_sec + (double)time.tv_usec * .000001);
+}
+
+/**
+ * @brief
+ *	Returns the current processor time.
+ *
+ * @return double - number of seconds.
+ **/
+static double
+get_cputime()
+{
+	clock_t	clock_cycles;
+	clock_cycles = clock();
+	if (clock_cycles == -1)
+		return (0);
+	return (double)clock_cycles / CLOCKS_PER_SEC;
```

Comment:
CLOCKS_PER_SEC should never be zero, but it would cause a failure if it were.

Reply:
Ok, I'll put an assert(CLOCKS_PER_SEC != 0) statement out there, to reflect that this is not expected.